utils/data.py

```diff
@@ -13,11 +13,21 @@ def __init__(self, *args, prefix=None, **kwargs):
         self.trivia = Trivia()
 
     async def on_message(self, msg):
-        if not self.is_ready() or msg.author.bot or not permissions.can_handle(msg, "send_messages"):
+        if not self.is_ready() or not permissions.can_handle(msg, "send_messages"):
             return
 
         await self.process_commands(msg)
 
+    ## @story{51} In order to test command coroutines, a second bot is needed.
```

Comment:
Why is a second bot needed?

Reply:
A second bot is needed to issue commands to the tested bot in a controlled test server. This is black-box integration testing but relatively easy to set up. In order to do white-box testing or testing without a connection to Discord, a highly complex set of mocks would be needed, beginning with the context object but also literally everything its methods could possibly return. There have been a couple of attempts at this that can be found on GitHub, but they are woefully incomplete, contain serious bugs, and are barely alpha quality at best. Creating our own mocks was out of the question given the time constraints we have and the level of complexity involved, so the only reasonable option was distest, which requires a second bot.